src/test/java/com/amihaiemil/camel/ReadPointedScalarTest.java

```diff
@@ -0,0 +1,215 @@
+     */
+    @Test
+    public void comparesToSequence() {
+        final List<YamlLine> lines = new ArrayList<>();
+        lines.add(new RtYamlLine("First Line", 1));
+        lines.add(new RtYamlLine("Second Line", 2));
+        lines.add(new RtYamlLine("Third Line", 3));
+        final ReadPointedScalar scalar = 
+            new ReadPointedScalar(new RtYamlLines(lines));
+        RtYamlSequence seq = new RtYamlSequence(new LinkedList<YamlNode>());
+        MatcherAssert.assertThat(scalar.compareTo(seq), Matchers.lessThan(0));
+    }
+    
+    /**
+     * ReadPointedScalar can compare itself to a Scalar.
+     */
+    @Test
+    public void comparesToScalar() {
+        final List<YamlLine> lines = new ArrayList<>();
+        lines.add(new RtYamlLine("Java", 1));
+        final ReadPointedScalar pipeScalar = 
+            new ReadPointedScalar(new RtYamlLines(lines));
+        final Scalar scalar = new Scalar("Java");
+        MatcherAssert.assertThat(pipeScalar.compareTo(scalar), Matchers.is(0));
+    }
+    
+    /**
+     * ReadPointerScalar can compare itself to other ReadPipeScalar.
+     */
+    @Test
+    public void comparesToReadPipeScalar() {
+        final List<YamlLine> lines = new ArrayList<>();
+        lines.add(new RtYamlLine("Java", 1));
+        final ReadPointedScalar first = 
+            new ReadPointedScalar(new RtYamlLines(lines));
+        final ReadPipeScalar second = 
+            new ReadPipeScalar(new RtYamlLines(lines));
+        MatcherAssert.assertThat(first.compareTo(second), Matchers.is(0));
+    }
+    
+    /**
+     * ReadPointerScalar can compare itself to other ReadPointedScalar.
+     */
+    @Test
+    public void comparesToReadPointedScalar() {
+        final List<YamlLine> lines = new ArrayList<>();
+        lines.add(new RtYamlLine("Java", 1));
+        final ReadPointedScalar first = 
+            new ReadPointedScalar(new RtYamlLines(lines));
+        final ReadPointedScalar second = 
+            new ReadPointedScalar(new RtYamlLines(lines));
+        MatcherAssert.assertThat(first.compareTo(second), Matchers.is(0));
+    }
+    
+    /**
+     * ReadPointedScalar can return value when all lines has no indentation.
+     */
+    @Test 
+    public void returnsValueWithNoIndentation() {
+        final List<YamlLine> lines = new ArrayList<>();
+        lines.add(new RtYamlLine("Mark McGwire's", 1));
+        lines.add(new RtYamlLine("year was crippled", 2));
+        lines.add(new RtYamlLine("by a knee injury.", 3));
+        final ReadPointedScalar scalar = 
+            new ReadPointedScalar(new RtYamlLines(lines));
+        MatcherAssert.assertThat(
+            scalar.value(),
+            Matchers.is("Mark McGwire's year was crippled by a knee injury.")
+        );
+    }
+    
+    /**
+     * ReadPointedScalar can return value when line has different indentation.
+     */
+    @Test
+    @Ignore
+    public void returnsValueWithSeveralIndentation() {
```

Comment:
@SherifWaly But why are you interested in the lines that have inner indentation? As I understand, you are not supposed to touch them (they keep their line endings). So it's okay that they are skipped, no? :D